### PR TITLE
feat: improve Dockerfile and remove vuln

### PR DESCRIPTION
**Pull request by julienlau**
improve Dockerfile in a security perspective:
- remove obsolete openjdk base image
- use node version that do not include critcal vulnerabilities
- specify UID >= 10000
- remove wget and curl (we may skip this step)

+ solve an issue due to *procps* not being installed
+ install atop basic monitoring tool

I checked on my environments that the image is functionnal

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ RUN wget "https://downloads.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPAR
 WORKDIR /home/app/frontend/
 RUN yarn install && yarn build
 
-FROM openjdk:11-jre-slim-bullseye
+FROM eclipse-temurin:11-jdk-jammy
```
Comment (Minutis) [inline Dockerfile:26]: Previous image was of JRE, maybe it would work with `eclipse-temurin:11-jre-jammy`?

File: Dockerfile [hunk 2]
```diff
@@ -45,7 +45,25 @@ COPY --from=frontend /home/app/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION
...
+    apt-get clean && rm -rf /var/lib/apt/lists /var/cache/apt/*
+
 EXPOSE 8080
 EXPOSE 25333
 
 ENTRYPOINT ["java", "-jar", "/home/app/application.jar"]
+
+# Specify the User that the actual main process will run as
+USER ${spark_uid}
+SHELL ["/bin/bash", "-c"]
```
Comment (Minutis) [inline Dockerfile:69]: Missing empty line in the end.

Comment (Minutis) [reply Dockerfile:69]: Also I do not think setting the shell is needed?

File: Dockerfile [hunk 3]
```diff
@@ -45,7 +45,25 @@ COPY --from=frontend /home/app/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION
 
 COPY k8s/ ./k8s/
 
+ARG spark_uid=10000
+ARG spark_gid=10001
+RUN groupadd -g ${spark_gid} spark && useradd spark -u ${spark_uid} -g ${spark_gid} -m -s /bin/bash
+RUN mkdir -p /home/db /tmp/spark-events /tmp/staging /tmp/s3a ${SPARK_HOME}/workdir && \
```
Comment (Minutis) [inline Dockerfile:51]: Why do we need to create all these `/home/db /tmp/spark-events /tmp/staging /tmp/s3a`? I do not think they are in use? And after that setting any permissions below seems unnecessary.

File: Dockerfile (hunk 1 above)
Comment (julienlau) [reply Dockerfile:26]: I also assumed that initially.
However, if you use JRE, the stage running `gradle` throw an exception saying that it cannot find java compiler

File: Dockerfile (hunk 2 above)
Comment (julienlau) [reply Dockerfile:69]: you're right it is not needed

File: Dockerfile (hunk 3 above)
Comment (julienlau) [reply Dockerfile:51]: Yes you're right, those directories are only needed in the spark image.
I used the /home/db directory to do some test using H2 on file system using a PVC for persistence on reboot.
It is not needed either

File: Dockerfile (hunk 1 above)
Comment (Minutis) [reply Dockerfile:26]: You are right. But in that case we could do the compiling in JDK and for runtime image use JRE here.

**Comment by julienlau** [comment]
Please find a new commit that I tested on my side and is OK.
It takes your review into account

File: Dockerfile (hunk 1 above)
Comment (julienlau) [reply Dockerfile:26]: ok I'll have a try

Comment (julienlau) [reply Dockerfile]: and it works and is included in the latest commit

**Review by Minutis — APPROVED** [review]
Looks good.

**Pull request merged**